src/test/java/com/parkhomovsky/bookstore/services/BookServiceTest.java

```diff
@@ -0,0 +1,198 @@
+            .setCoverImage("URL");
+    private static final List<BookDtoWithoutCategoryIds> ALL_FICTION_BOOKS_DTO
+            = List.of(RESPONSE_EXIST_TEST_BOOK_DTO_WITHOUT_CATEGORIES);
+    private static final Long ABSENT_BOOK_ID = 128L;
+    private static final List<BookDto> ALL_BOOKS_DTO_RESPONSE =
+            List.of(RESPONSE_EXIST_TEST_BOOK_DTO, RESPONSE_CREATED_DOTA_BOOK_DTO);
+    private static final List<Book> ALL_BOOKS = List.of(EXIST_TEST_BOOK, CREATED_DOTA_BOOK);
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private BookMapper bookMapper;
+    @Mock
+    private CategoryRepository categoryRepository;
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @Test
+    @DisplayName("Verify successful creating book with a valid input")
+    public void create_validBookParameters_shouldReturnBookDto() {
+        when(bookMapper.toEntity(REQUEST_CREATE_DOTA_BOOK_DTO)).thenReturn(CREATED_DOTA_BOOK);
+        when(bookRepository.save(CREATED_DOTA_BOOK)).thenReturn(CREATED_DOTA_BOOK);
+        when(bookMapper.toDto(CREATED_DOTA_BOOK)).thenReturn(RESPONSE_CREATED_DOTA_BOOK_DTO);
+
+        BookDto actual = bookService.create(REQUEST_CREATE_DOTA_BOOK_DTO);
+        assertEquals(RESPONSE_CREATED_DOTA_BOOK_DTO, actual);
+    }
+
+    @Test
+    @DisplayName("Verify returned book with a valid input id")
+    public void findById_validBookId_shouldReturnBookDto() {
+        when(bookMapper.toDto(EXIST_TEST_BOOK))
+                .thenReturn(RESPONSE_EXIST_TEST_BOOK_DTO);
+        when(bookRepository.findById(EXIST_TEST_BOOK.getId()))
+                .thenReturn(Optional.of(EXIST_TEST_BOOK));
+        BookDto actual = bookService.getById(EXIST_TEST_BOOK.getId());
+        assertEquals(RESPONSE_EXIST_TEST_BOOK_DTO, actual);
+    }
+
+    @Test
+    @DisplayName("Verify returned book with a wrong input id")
+    public void findById_invalidBookId_shouldThrowEntityNotFoundException() {
+        when(bookRepository.findById(ABSENT_BOOK_ID)).thenReturn(Optional.empty());
+
+        Exception exception = assertThrows(
+                EntityNotFoundException.class,
+                () -> bookService.getById(ABSENT_BOOK_ID)
+                );
+        String expectedMessage = "Can`t find any book with id: "
+                + ABSENT_BOOK_ID
+                + " during getById";
+        String actualMessage = exception.getMessage();
+        assertEquals(expectedMessage, actualMessage);
+    }
+
+    @Test
+    @DisplayName("Verify all returned books with creating another one")
+    public void getAll_validBookParams_shouldReturnListBookDto() {
+        when(bookMapper.toEntity(REQUEST_CREATE_DOTA_BOOK_DTO)).thenReturn(CREATED_DOTA_BOOK);
+        when(bookRepository.save(CREATED_DOTA_BOOK)).thenReturn(CREATED_DOTA_BOOK);
+        when(bookMapper.toDto(CREATED_DOTA_BOOK)).thenReturn(RESPONSE_CREATED_DOTA_BOOK_DTO);
+        when(bookMapper.toDto(EXIST_TEST_BOOK)).thenReturn(RESPONSE_EXIST_TEST_BOOK_DTO);
```

Comment:
try to separate preparing logic, execution logic, and assertion logic by empty lines, would be easier to read